src/cdk/tree/control/nested-tree-control.ts

```diff
@@ -25,16 +25,17 @@ export class NestedTreeControl<T> extends BaseTreeControl<T> {
    */
   expandAll(): void {
     this.expansionModel.clear();
-    let toBeExpanded = <any>[];
-    this.dataNodes.forEach(dataNode => toBeExpanded.push(...this.getDescendants(dataNode)));
-    this.expansionModel.select(...toBeExpanded);
+    const allNodes = this.dataNodes.map(dataNode => [...this.getDescendants(dataNode), dataNode])
+        .reduce((first, second) => [...first, ...second]);
```

Comment:
I don't think you need both `map` and `reduce` here. Something like this should work the same:

```ts
this.dataNodes.reduce((accumulator, dataNode) => [...accumulator, ...this.getDescendants(dataNode), dataNode], []);
```